### PR TITLE
Add SLJIT_UTIL_SIMPLE_STACK_ALLOCATION flag to use normal allocator for

### PR DESCRIPTION
**Pull request by cohomology**
Add SLJIT_UTIL_SIMPLE_STACK_ALLOCATION flag to use SLJIT_MALLOC/SLJIT_FREE to allocate/free the stacj

### CODE REVIEW

File: sljit_src/sljitUtils.c [hunk 1]
```diff
@@ -316,7 +330,7 @@ SLJIT_API_FUNC_ATTRIBUTE sljit_u8 *SLJIT_FUNC sljit_stack_resize(struct sljit_st
 				return NULL;
 		}
 	}
-#else
+#else
```
Comment (zherczeg) [inline sljit_src/sljitUtils.c]: Please remove these extra spaces.

File: sljit_src/sljitUtils.c [hunk 2]
```diff
@@ -206,22 +206,34 @@ static SLJIT_INLINE sljit_s32 open_dev_zero(void)
...
+	{
+		SLJIT_FREE(stack, allocator_data);
+		return NULL;
+	}
+	stack->min_start = (sljit_u8 *)ptr;
+ 	stack->end = stack->min_start + max_size;
+ 	stack->start = stack->end - start_size; 
+#else // SLJIT_UTIL_SIMPLE_STACK_ALLOCATION
 #ifdef _WIN32
+	SYSTEM_INFO si;
```
Comment (zherczeg) [inline sljit_src/sljitUtils.c]: Declaration after code is supported by msvc? Is it c99 compatible? I know some compilers complain about this, but I am not sure which standard introduced them.

**Comment by cohomology** [comment]
As there where too many #ifdefs for my taste and the shared code was not that much, I factored out the "simple" cases. I also got rid of a warning, when compiling with "-std=c89" on gcc. 
I did not test yet with msvc, but I can do this tomorrow.

File: sljit_src/sljitConfig.h [hunk 3]
```diff
@@ -69,12 +69,19 @@ extern "C" {
...
+/* Implements a stack like data structure (by using mmap / VirtualAlloc  */
+/* or a custom allocator). */
 #ifndef SLJIT_UTIL_STACK
 /* Enabled by default */
 #define SLJIT_UTIL_STACK 1
 #endif
 
+/* Uses user provided allocator to allocate the stack (see SLJIT_UTIL_STACK) */
+#ifndef SLJIT_UTIL_SIMPLE_STACK_ALLOCATION
+/* disabled by default */
```
Comment (zherczeg) [inline sljit_src/sljitConfig.h]: Use capital D: Disabled.

**Comment by cohomology** [comment]
Hi,

I changed the casing.

Can you bring the change to PCRE?

Thanks,
Kilian.

**Comment by zherczeg** [comment]
Will do it.

**Comment by zherczeg** [comment]
Done.

**Pull request merged**